components/scream/src/share/io/scorpio_output.cpp

```diff
@@ -1,102 +1,143 @@
   register_views();
 
-  // If this is a restart run that requires a restart history file read input here:
-  if (m_read_restart_hist)
-  {
+  // If this is "normal" output of a restarted run, and avg_type requires restart data,
+  // then we have to open the history restart file and read its data.
+  // The user can skip this (which means the Output averaging would start from scratch)
+  // by setting "Restart History" to false.
+  m_is_restarted_run &= m_params.get("Restart History",true);
+  if (m_has_restart_data && m_is_restarted_run && !m_is_model_restart_output) {
     std::ifstream rpointer_file;
     rpointer_file.open("rpointer.atm");
     std::string filename;
     bool found = false;
-    std::string testname = m_casename+"."+m_avg_type+"."+m_out_units+"_x"+std::to_string(m_out_frequency);
-    while (rpointer_file >> filename)
-    {
-      if (filename.find(testname) != std::string::npos)
-      {
+    std::string testname;
+    if (m_params.isParameter("RESTART HISTORY FILENAME")) {
+      auto hist_fname = m_params.get<std::string>("RESTART HISTORY FILENAME");
+      testname = compute_filename_root(hist_fname);
+    } else {
+      testname = compute_filename_root(m_casename);
+    }
+    while (rpointer_file >> filename) {
+      if (filename.find(testname) != std::string::npos) {
         found = true;
         break;
       }
     }
-    if ( not found )
-    {
-      printf("Warning! No restart history file found in rpointer file for %s, using current values in field manager\n",m_casename.c_str());
-    }
-    // Register rhist file as input and copy data to local views
-    ekat::ParameterList res_params("Input Parameters");
-    res_params.set<std::string>("FILENAME",filename);
-    res_params.set<std::string>("GRID","Physics");
-    res_params.set<bool>("RHIST",true);
-    auto& f_list = res_params.sublist("FIELDS");
-    Int fcnt = 1;
-    f_list.set<Int>("Number of Fields",m_fields.size());
-    for (auto name : m_fields)
-    {
-      f_list.set<std::string>("field "+std::to_string(fcnt),name);
-      fcnt+=1;
-    }
-    input_type rhist_in(m_comm,res_params,m_field_mgr,m_grid_mgr);
-    rhist_in.init();
-    for (auto name : m_fields)
-    {
-      auto l_view = rhist_in.pull_input(name);
-      m_host_views_1d.at(name) = l_view;
+
+    // If the history restart file is not found, we must HOPE that it is because
+    // the last model restart step coincided with a model output step, in which case
+    // a restart history file is not written.
+    // TODO We have NO WAY of checking this from this class, but the OutputManager
+    //      *might* be able to figure out if this is the case, and perhaps set
+    //      "Restart History" to false in the input parameter list.
+    //      For now, simply print a warning.
+    if (found) {
+      // If restart data is needed, the history restart file *must* be found.
+      EKAT_REQUIRE_MSG(found,
```

Comment:
`found` is always `true` in this context.

Reply:
Good catch.